### PR TITLE
node:http2: saturate numeric session options instead of wrapping to the minimum

### PR DESCRIPTION
**Pull request by robobun**
### Problem
- `http2.createServer({ maxSessionMemory: Number.MAX_SAFE_INTEGER })` refuses the second request stream with `ENHANCE_YOUR_CALM` (rstCode 11) once about 1MB of response data is queued. Node serves it. Any value at or above 2^51, and exactly 2^32, fails the same way. `@grpc/grpc-js` 1.14 passes `Number.MAX_SAFE_INTEGER` to disable the limit, so Pulumi programs under Bun fail with `RESOURCE_EXHAUSTED`.
- Two truncations stack. `JSC__JSValue__toUInt64NoTruncate` (`src/jsc/bindings/bindings.cpp`) routed doubles through `tryConvertToInt52` and returned 0 for a double at or above 2^51. `src/runtime/api/bun/h2_frame_parser.rs` then cast with `as u32`, which wraps 2^32 to 0. Both paths land on the `.max(1)` floor: a 1MB budget.

### Fix
- `toUInt64NoTruncate` now uses `JSC::toUInt64` for doubles, with one clamp in front: values at or above 2^64, including `+Infinity`, give `UINT64_MAX`. Below that the conversion is exact for `[0, 2^64)`, `NaN` and `-Infinity` give 0, and negatives wrap the same way the int32 path always has (`-1` and `-1.5` both give `UINT64_MAX`). No UB from `static_cast<uint64_t>(double)`.
- This is the function behind every Rust `to_uint64_no_truncate()` caller, so `Bun.hash` seeds and the other call sites get the same fix.
- The http2 parser narrows the u64 into its u32 limit fields with `u32::try_from(..).unwrap_or(u32::MAX)` instead of `as u32`. This covers `maxSessionMemory` and the five sibling options read in the same block.

### Tests
- `test/js/node/http2/h2-conformance.test.ts`: three new cases (`Number.MAX_SAFE_INTEGER`, `2**51`, `2**32`) serve the second stream.
- `test/js/bun/util/hash.test.js`: Number seeds in `[2^51, 2^64)` match their BigInt equivalents; `2**64`, `Infinity`, `-1`, `-1.5` equal seed `2^64-1`; `NaN`, `-Infinity` equal seed 0.

Fixes #41294

### CODE REVIEW

**Review by Jarred-Sumner — COMMENTED** [review]
maybe instead to_uint64_no_truncate should be fixed

**Comment by robobun** [comment]
Agreed. I am reworking this to fix `toUInt64NoTruncate` in bindings.cpp so a double at or above 2^51 no longer reads as 0. The parser still needs a saturating u64 to u32 narrowing for 2^32, which wrapped to 0 on its own. Will push shortly.

File: src/jsc/bindings/bindings.cpp [hunk 1]
```diff
@@ -4430,14 +4430,15 @@ uint64_t JSC__JSValue__toUInt64NoTruncate(JSC::EncodedJSValue val)
...
-    if (result != JSC::JSValue::notInt52) {
-        if (result < 0)
-            return 0;
-
-        return static_cast<uint64_t>(result);
-    }
-    return 0;
+    // Saturate: NaN and negatives read as 0, values at or above 2^64 read as
+    // UINT64_MAX. Callers pass Number.MAX_SAFE_INTEGER (or Infinity) to mean
+    // "no limit", so a huge double must not collapse to 0.
```
[the remark this answers is not in the document]
Comment (robobun) [reply src/jsc/bindings/bindings.cpp]: Not a workaround, but the comment was longer than needed. Cut to one line in d61d937191.

File: src/runtime/api/bun/h2_frame_parser.rs [hunk 2]
```diff
@@ -7521,6 +7521,12 @@ impl H2FrameParser {
         self.native_socket.detach();
     }
 
+    /// Session limits are u32 fields. Saturate so a huge value (grpc-js passes
+    /// `Number.MAX_SAFE_INTEGER` to mean "no limit") does not wrap to 0.
```
[the remark this answers is not in the document]
Comment (robobun) [reply src/runtime/api/bun/h2_frame_parser.rs]: Not a workaround, but the comment was longer than needed. Cut to one line in d61d937191.

**Comment by robobun** [comment]
Addressed the negative int32 finding in 7c6d47c419. A negative int32 now reads as 0, the same as a negative double. The hash test covers -1 next to -1.5.

File: src/jsc/bindings/bindings.cpp [hunk 3]
```diff
@@ -4426,18 +4427,15 @@ uint64_t JSC__JSValue__toUInt64NoTruncate(JSC::EncodedJSValue val)
     }
 
     if (value.isInt32()) {
-        return static_cast<uint64_t>(value.asInt32());
+        return static_cast<uint64_t>(static_cast<int64_t>(value.asInt32()));
     }
```
[the remark this answers is not in the document]
Comment (robobun) [reply src/jsc/bindings/bindings.cpp:4431]: Wrapping for negatives is the intended contract as of 46b47e1e06. It matches the int32 path this function always had, and the PR body now states it. The test encodes that on purpose.
